### DISCUSSION TITLE
Allow changing product type after product creation

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/commerce#2430] [Ideas]
Just like we can switch entry type for entry, we should be able to switch product type for product. 

Maybe there should be separate product types and product field layouts, so we would be able to change fields, without changing URL settings. Like we can switch entries entry type without changing their section.

Use case:
when importing products from feedme, we sometimes cannot determine which type it should belong to, we should have ability to set it manually later.

**Comment by lukeholder** [comment]
The problem is that switching product types changes more than just the custom fields, there are a number of rules that are also affected (like allowing multiple variants etc, for which the product edit screen is not equiped to change dynamically).

Having said that you can change the product type in code as long as you resave the products afterwards, and make them valid for the new product type before saving.

We don't plan to add product type/entry type like switching to products at the moment, but will keep this request open.

**Comment by nitech** [comment]
I'll explain why this is something I also need: 

We built our site in multiple phases. In phase 1 we might not be able to see the needs for phase 2 or 3. Right now we're in phas 1, and we have the following product types (listing some of them): 

- Sewing Machines
- Compressors
- LED lights
- Other products

`Other products` typically contain products we only have one or two of- where it seems wrong to create a specific product type for just one product. One example is "Sewing Thread", which in phase 2 will be perhaps 20 different products with each 100-200 variants.

In other words, the conversion from one product type to another is typically from less to more complex.

Or am I doing it wrong? Should I have one product type for each product regardless of whether there is only one single product?
